tools/azure-pipelines/build-apache-repo.yml

```diff
@@ -0,0 +1,94 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#
+# This file defines the Flink build for the "apache/flink" repository, including
+# the following:
+#  - PR builds
+#  - custom triggered e2e tests
+#  - nightly builds
+
+
+
+schedules:
+- cron: "0 0 * * *"
+  displayName: Daily midnight build
+  branches:
+    include:
+    - master
+  always: true # run even if there were no changes to the mentioned branches
+
+resources:
+  containers:
+  # Container with Maven 3.2.5, SSL to have the same environment everywhere.
+  - container: flink-build-container
+    image: rmetzger/flink-ci:ubuntu-jdk8-amd64-e005e00
+
+
+variables:
+  MAVEN_CACHE_FOLDER: $(Pipeline.Workspace)/.m2/repository
+  MAVEN_OPTS: '-Dmaven.repo.local=$(MAVEN_CACHE_FOLDER)'
+  CACHE_KEY: maven | $(Agent.OS) | **/pom.xml, !**/target/**
+  CACHE_FALLBACK_KEY: maven | $(Agent.OS)
+  CACHE_FLINK_DIR: $(Pipeline.Workspace)/flink_cache
+
+stages:
+  # CI / PR triggered stage:
+  - stage: ci_build
+    displayName: "CI Build (custom builders)"
+    condition: not(eq(variables['Build.Reason'], in('Schedule', 'Manual')))
+    jobs:
+      - template: jobs-template.yml
+        parameters:
+          stage_name: ci_build
+          test_pool_definition:
+            name: Default
+          e2e_pool_definition:
+            vmImage: 'ubuntu-latest'
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11"
+
+  # Special stage for midnight builds:
+  - stage: cron_build_on_azure_os_free_pool
+    displayName: "Cron build on free Azure Resource Pool"
+    dependsOn: [] # depending on an empty array makes the stages run in parallel
+    condition: or(eq(variables['Build.Reason'], 'Schedule'), eq(variables['MODE'], 'nightly'))
+    jobs:
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_default
+          test_pool_definition:
+            vmImage: 'ubuntu-latest'
+          e2e_pool_definition:
+            vmImage: 'ubuntu-latest'
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.11"
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_scala2_12
+          test_pool_definition:
+            vmImage: 'ubuntu-latest'
+          e2e_pool_definition:
+            vmImage: 'ubuntu-latest'
+          environment: PROFILE="-Dhadoop.version=2.8.3 -Dinclude_hadoop_aws -Dscala-2.12 -Phive-1.2.1"
+      - template: jobs-template.yml
+        parameters:
+          stage_name: cron_build_jdk11
```

Comment:
where are we actually setting up jdk11?

Reply:
Yeah, no. This is some left over work in progress.
What do you prefer: adding proper jdk11 support as part of this PR, or removing it now, and doing it in a separate PR? (I prefer the latter)

Reply:
separate PR, java 11 runs only on cron jobs anyway which are a follow-up in any case.

Reply:
separate PR

Reply:
ok, thx for the fast response.

Reply:
https://issues.apache.org/jira/browse/FLINK-15834

Reply:
ping to remove the jdk11 stage